gluten-substrait/src/main/scala/org/apache/gluten/execution/ValidatablePlan.scala

```diff
@@ -63,13 +65,26 @@ trait ValidatablePlan extends GlutenPlan with LogLevelUtil {
    * Validate whether this SparkPlan supports to be transformed into substrait node in Native Code.
    */
   final def doValidate(): ValidationResult = {
-    val schemaValidationResult = BackendsApiManager.getValidatorApiInstance
-      .doSchemaValidate(schema)
-      .map {
-        reason =>
-          ValidationResult.failed(s"Found schema check failure for $schema, due to: $reason")
+    val schemaValidationResult =
+      try {
+        BackendsApiManager.getValidatorApiInstance
+          .doSchemaValidate(schema)
+          .map {
+            reason =>
+              ValidationResult.failed(
+                s"Found schema check failure for $schema, due to: $reason")
+          }
+          .getOrElse(ValidationResult.succeeded)
+      } catch {
+        case e: AnalysisException =>
```

Comment:
Suggest to use `UnsolvedException`, 1) it's more specific 2) To make this exception catching also works Spark 4.0.

Prior to Spark 4.0, UnsolvedException extends AnalysisException, but since Spark 4.0, it extends SparkException.

Reply:
@PHILO-HE Yes. Updated.